### ISSUE TITLE
client.images: 'show only images belonging to repository name' no longer working

### ISSUE THREAD

**Issue by pommetjehorlepiep** [docker/docker-py#2730]
- docker-py: 4.4.0 (on Arch Linux)
- Low-level API `images` command:
     parameter `name` seems to be completely ignored and the full list of images is returned as result.

Expected behaviour:
=> a dict or list of images which belong to repository specified with parameter `name`

Actual behaviour:
=> a dict or list of all images, although a specific repository was specified with parameter `name`

![image](https://user-images.githubusercontent.com/8982487/102768609-2a2ed100-43cd-11eb-9b84-01dac7c0218a.png)
![image](https://user-images.githubusercontent.com/8982487/102768996-bfca6080-43cd-11eb-8eaf-57c26459b015.png)

Docker version information:
```
community/python-docker 4.4.0-1 (196.1 KiB 1.3 MiB) (Installed)
    A Python library for the Docker Engine API

Client:
 Version:           20.10.1
 API version:       1.41
 Go version:        go1.15.6
 Git commit:        831ebeae96
 Built:             Tue Dec 15 22:25:01 2020
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.1
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.15.6
  Git commit:       f0014860c1
  Built:            Tue Dec 15 22:24:28 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b.m
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Comment by william-richard** [comment]
I noticed this too - is it possible this is the same issue? #2722

**Comment by pommetjehorlepiep** [comment]
> I noticed this too - is it possible this is the same issue? #2722

**Comment by pommetjehorlepiep** [comment]
Oops...didn't mean to close it

**Comment by pommetjehorlepiep** [comment]
> I noticed this too - is it possible this is the same issue? #2722

@william-richard It certainly looks like it.

**Comment by thaJeztah** [comment]
this should be fixed by https://github.com/docker/docker-py/pull/2708 indeed, which is included in 4.4.1

**Comment by pommetjehorlepiep** [comment]
@thaJeztah, I've verified that the issue has been resolved in 4.4.1. Filtering by `name` works again as intended!

Closing issue
